ambry-api/src/main/java/com/github/ambry/router/KeyManagementService.java

```diff
@@ -47,21 +48,23 @@
   /**
    * Fetches the key associated with the pair of AccountId and ContainerId. User is expected to have
    * registered using {@link #register(short, short)} for this pair before fetching keys.
+   * @param restRequest the {@link RestRequest} to use
    * @param accountId refers to the id of the {@link Account} for which key is expected
    * @param containerId refers to the id of the {@link Container} for which key is expected
    * @return T the key associated with the accountId and containerId
    * @throws {@link GeneralSecurityException} on KMS unavailability or if key is not registered
    */
-  T getKey(short accountId, short containerId) throws GeneralSecurityException;
+  T getKey(RestRequest restRequest, short accountId, short containerId) throws GeneralSecurityException;
```

Comment:
Let's document that a KMS impl should be prepared to handle the case where `restRequest` is null.